### PR TITLE
Json streaming support [WIP]

### PR DESCRIPTION
**Pull request by gabfssilva**
Marking it as still WIP since I'm going to implement for all modules. I just need to know if I should fix something in this module (play-json) first.

=)

gif

### CODE REVIEW

**Comment by hseeberger** [comment]
Thanks for your contribution. I am currentlyon vacation and will review as soon as possible when back.

File: akka-http-play-json/src/main/scala/de/heikoseeberger/akkahttpplayjson/PlayJsonSupport.scala [hunk 1]
```diff
@@ -85,4 +123,61 @@ trait PlayJsonSupport {
...
   ): ToEntityMarshaller[A] =
     jsonStringMarshaller.compose(printer).compose(writes.writes)
+
+  /**
+    * `ByteString` => `A`
+    *
+    * @tparam A type to decode
+    * @return unmarshaller for any `A` value
+    */
+  implicit def fromByteStringUnmarshaller[A: Reads]: Unmarshaller[ByteString, A] = Unmarshaller {
```
Comment (hseeberger) [inline akka-http-play-json/src/main/scala/de/heikoseeberger/akkahttpplayjson/PlayJsonSupport.scala]: Minor formatting issue: only signature on firtst line.
Move code after `=` on next line, pleasen.

File: akka-http-play-json/src/main/scala/de/heikoseeberger/akkahttpplayjson/PlayJsonSupport.scala [hunk 2]
```diff
@@ -85,4 +123,61 @@ trait PlayJsonSupport {
...
+    *
+    * @tparam A type to decode
+    * @return unmarshaller for `Source[A, _]`
+    */
+  implicit def sourceUnmarshaller[A: Reads](
+      implicit support: JsonEntityStreamingSupport = EntityStreamingSupport.json()
+  ): FromEntityUnmarshaller[SourceOf[A]] =
+    Unmarshaller
+      .withMaterializer[HttpEntity, SourceOf[A]] { implicit ec => implicit mat => entity =>
+        def asyncParse(bs: ByteString): Future[A] =
```
Comment (hseeberger) [inline akka-http-play-json/src/main/scala/de/heikoseeberger/akkahttpplayjson/PlayJsonSupport.scala]: Do we need the type annotations for the local methods?

File: akka-http-play-json/src/main/scala/de/heikoseeberger/akkahttpplayjson/PlayJsonSupport.scala [hunk 3]
```diff
@@ -85,4 +123,61 @@ trait PlayJsonSupport {
...
+    * @tparam A type to encode
+    * @return marshaller for any `SourceOf[A]` value
+    */
+  implicit def sourceMarshaller[A](
+      implicit writes: Writes[A],
+      printer: JsValue => String = Json.stringify,
+      support: JsonEntityStreamingSupport = EntityStreamingSupport.json()
+  ): ToEntityMarshaller[SourceOf[A]] =
+    jsonSourceStringMarshaller
+      .compose(jsonSource[A])
```
Comment (hseeberger) [inline akka-http-play-json/src/main/scala/de/heikoseeberger/akkahttpplayjson/PlayJsonSupport.scala]: No new line needed here.

**Comment by gabfssilva** [comment]
Finished 6 modules, three to go.

**Comment by gabfssilva** [comment]
I left `akka-http-avsystem-gencodec` behind since it's not compiling.

I think that's it. Sorry for the 1k LoC, but, the code is pretty much the same between the modules, with a few adjusts to work with the respective libraries.

File: akka-http-argonaut/src/test/scala/de/heikoseeberger/akkahttpargonaut/ExampleApp.scala [hunk 4]
```diff
@@ -18,12 +18,16 @@ package de.heikoseeberger.akkahttpargonaut
...
 import akka.http.scaladsl.server.Directives
-import akka.stream.Materializer
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.scaladsl.Source
 import argonaut.Argonaut.casecodec1
 import argonaut.CodecJson
+
 import scala.concurrent.Await
-import scala.concurrent.duration.Duration
+import scala.concurrent.duration._
```
Comment (hseeberger) [inline akka-http-argonaut/src/test/scala/de/heikoseeberger/akkahttpargonaut/ExampleApp.scala]: No wildcard imports.

File: akka-http-argonaut/src/test/scala/de/heikoseeberger/akkahttpargonaut/ExampleApp.scala [hunk 5]
```diff
@@ -18,12 +18,16 @@ package de.heikoseeberger.akkahttpargonaut
...
 import akka.http.scaladsl.Http
+import akka.http.scaladsl.marshalling.Marshal
+import akka.http.scaladsl.model.{ HttpRequest, RequestEntity }
 import akka.http.scaladsl.server.Directives
-import akka.stream.Materializer
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.scaladsl.Source
 import argonaut.Argonaut.casecodec1
 import argonaut.CodecJson
+
```
Comment (hseeberger) [inline akka-http-argonaut/src/test/scala/de/heikoseeberger/akkahttpargonaut/ExampleApp.scala]: No blank lines => Please fix your IntelliJ (?) settings

File: akka-http-argonaut/src/test/scala/de/heikoseeberger/akkahttpargonaut/ExampleApp.scala [hunk 6]
```diff
@@ -43,7 +47,7 @@ object ExampleApp {
     Await.ready(system.terminate(), Duration.Inf)
   }
 
-  def route(implicit mat: Materializer) = {
+  def route(implicit sys: ActorSystem) = {
```
Comment (hseeberger) [inline akka-http-argonaut/src/test/scala/de/heikoseeberger/akkahttpargonaut/ExampleApp.scala]: s/sys/system

**Pull request merged**